andrepereiradasilva/joomla-cms
# new article

Pull Request for Improvement.
### Summary of Changes

This PR intends to do several things in the article modal field:
- The most important, add a new button to create a new article directly (ex: when creating a menu item single article type)
  This can be very useful since, qith this, you can create a New article directly when you are creating a menu item.

Also:
- Add the ability to choose whether to show or not the edit, create, select, or clear buttons.
- Change the buttons behaviour. When no article is selected you have "Select" and "Create", when an article is selected you have "Edit" and "Clear".
- Simplify js in modals by moving most js to a new js file. Instead of inline js.
#### Some pictures

![image](https://cloud.githubusercontent.com/assets/9630530/18037743/6207df9c-6d83-11e6-9b55-2e576ac449fc.png)

![image](https://cloud.githubusercontent.com/assets/9630530/18037744/65f85e1a-6d83-11e6-8668-4d3a1594e5ad.png)

![image](https://cloud.githubusercontent.com/assets/9630530/18037750/77c1773a-6d83-11e6-95d6-68e558ab2f31.png)

![image](https://cloud.githubusercontent.com/assets/9630530/18037758/a4d1f358-6d83-11e6-905e-f2374df411d1.png)

If this is accepted i can do the same for contacts, categories, etc modals.
### Testing Instructions
1. Use latest staging and apply this patch
2. All article modal fields should work (menu item single article, user profile plugin tos, tinymce select article, article associations).
3. Test the new "Create" button
4. Code review to check all is ok.
### Documentation Changes Required

Don't know if exist any related to modal fields.
### B/C

I tried to make this all B/C (didn't remove now obsolete code in article modal.php).
Should be B/C but please check if i didn't forget anything.
